File: Greedy/2566.Maximum Difference by Remapping a Digit/Solution.java
class Solution {
    public int minMaxDifference(int num) {
        int mx = num;
        String s = String.valueOf(num);
        for (char c : s.toCharArray()) {
            if (c != '9') {
                mx = Integer.valueOf(String.valueOf(num).replace(c, '9'));
                break;
            }
        }
        return mx - Integer.valueOf(s.replace(s.charAt(0), '0'));
    }
}

File: Greedy/2566.Maximum Difference by Remapping a Digit/Solution.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        s = str(num)
        mx = num
        for c in s:
            if c != '9':
                mx = int(str(num).replace(c, '9'))
                break
        s = s.replace(s[0], '0')
        return mx - int(s)
